### PR TITLE
Upgrade to rails 6.1

### PR DESCRIPTION
**Pull request by bfcoder**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -11,13 +11,13 @@ git_source(:github) do |repo_name|
...
 # Bundle edge Rails instead: gem "rails", github: "rails/rails"
-gem "rails", "6.0.3.6"
+gem "rails", "6.1.3.1"
 
 # Improve boot time
 gem "bootsnap", require: false
 
 # Database
-gem "composite_primary_keys"
+gem "composite_primary_keys", git: "https://github.com/composite-primary-keys/composite_primary_keys.git" # unreleased rails 6.1 support
```
Comment (jbasdf) [inline Gemfile:20]: Are we OK with having an unreleased gem or do we need to wait for an official release?

File: app/models/user.rb [hunk 2]
```diff
@@ -39,7 +39,7 @@ def self.create_on_tenant(application_instance, user)
       user_dup = User.find_or_initialize_by(
         lti_user_id: user.lti_user_id,
       )
-      user_dup.update_attributes(user.copy_attributes)
+      user_dup.update(user.copy_attributes)
```
Comment (jbasdf) [inline app/models/user.rb:42]: Should we change this to update! ? I realize the old code ignored errors but maybe it shouldn't?

File: config/environments/development.rb [hunk 3]
```diff
@@ -56,12 +63,18 @@
...
+  # Annotate rendered view with file names.
+  # config.action_view.annotate_rendered_view_with_filenames = true
 
   # Use an evented file watcher to asynchronously detect changes in source code,
   # routes, locales, etc. This feature depends on the listen gem.
   config.file_watcher = ActiveSupport::EventedFileUpdateChecker
 
+  # Uncomment if you wish to allow Action Cable access from any origin.
+  # config.action_cable.disable_request_forgery_protection = true
+
```
Comment (jbasdf) [inline config/environments/development.rb:77]: Do we need to add config.hosts << ".#{ENV['APP_ROOT_DOMAIN']}" here as well? Perhaps it should be in application.rb?

File: config/initializers/filter_parameter_logging.rb [hunk 4]
```diff
@@ -1,4 +1,6 @@
 # Be sure to restart your server when you modify this file.
 
 # Configure sensitive parameters which will be filtered from the log file.
-Rails.application.config.filter_parameters += [:password]
+Rails.application.config.filter_parameters += [
+  :passw, :secret, :token, :_key, :crypt, :salt, :certificate, :otp, :ssn
```
Comment (jbasdf) [inline config/initializers/filter_parameter_logging.rb:5]: should this list also include :password?

File: config/environments/development.rb (hunk 3 above)
Comment (bfcoder) [reply config/environments/development.rb:77]: I moved it to its own initializer file.

File: config/initializers/filter_parameter_logging.rb (hunk 4 above)
Comment (bfcoder) [reply config/initializers/filter_parameter_logging.rb:5]: yeah, probably. I'll add it back in.

File: Gemfile (hunk 1 above)
Comment (bfcoder) [reply Gemfile:20]: We've had an unreleased gem in the past while we wait for an official release. I don't know that we can wait as I need rails 6.1 to continue some of my work downstream.

File: app/models/user.rb (hunk 2 above)
Comment (bfcoder) [reply app/models/user.rb:42]: hmm, I'll leave that decision up to you. It should probably be its own PR if we make that kind of change though.

File: config/initializers/filter_parameter_logging.rb (hunk 4 above)
Comment (bfcoder) [reply config/initializers/filter_parameter_logging.rb]: Done

**Review by jbasdf — APPROVED** [review]

**Comment by conradbeach** [comment]
🎉 Rails 6!

**Pull request merged**